Feature - Housing Level Overlay

rneeft opened the pull request:
**Summary**
This PR adds a new option to the Overlay menu for housing levels, addressing #1492 

**Details**

- Introduces a housing level overlay in the menu.
- Functionality is implemented and tested with the first eight housing types. 
[DONE] - **Translations**: Still missing. Currently only added in `english.c`. 
  - I would appreciate guidance on where to find the official names for the different housing levels in other languages.

**Next steps**
[DONE] - Add translations once I know where to reference the housing level names.
- Update the PR based on reviewer feedback


**Screenshot**
<img width="778" height="1195" alt="Screenshot From 2025-09-15 23-05-27-min" src="https://github.com/user-attachments/assets/057215a0-277c-4d98-b8f9-86789bababb4" />

rneeft committed:
first version

rneeft committed:
Added all housing levels to the overlay

rneeft commented:
@Johannes-callidus If `lang_get_building_type_string` only returns the singular, I think this is something you (and the other maintainers) need to decide:  

- Use the singular names directly in the UI, **or**  
- Keep the current approach with plural names stored in the language files.  

I’m happy to adjust either way — please let me know which you prefer.

rneeft committed:
Refactor the ADDITIONAL_OVERLAY_TR  to have everthing on one line like before

Johannes-callidus commented:
I think it's fine to just use plural custom translations, as the menu is generally very inconsistent (e.g. "Schools" Pl. but "Amphiteather" Sg.) and uses the original translations for the overlays unless it's an overlay added by augustus. The other overlays also don't use lang_get_building_type_string so it's okay to not do so as well.

rneeft committed:
Add translations of the new plural housing level overlay

rneeft commented:
Perfect, I’ve added translations for all the other languages  
(thanks to Copilot — I prompted it to translate the housing levels while keeping the Caesar 3 style intact).  

I’ve tested the overlays up to and including **Small Insulae**.  
Could you clarify what level of testing you expect before this PR can be considered complete?

Sephirex95 commented:
Happy to review that today but I have three requests:
1) Could you please remove translations apart from your native language and English? We generally tend to stick to our mother tongues for singular commits and request a 'missing translations update' from our translators for individual languages. While AI translations do make our life easier, they sometimes leave behind typos and mistranslations that stay in the files for longer than if we waited for a human 🤭 
2) I would suggest using already existing singular strings - as Johannes has mentioned, we've got some inconsistencies and will likely fix them in the future with an overhaul, so for now - less new `TR` entries to keep track of :) 
3) On the first look, well done with adding everything in it's place, but we're slowly trying to move away from static UI -> I've been fighting the codebase making dynamic elements where possible. Could you try (if not, that's alright - we'll target it later), and write some helpers to make the sub-buttons for the housing overlay dynamically positioned? so that if we remove or add some, we don't break the design? I can already imagine some complainers who would rather have them in groups like 'tents' 'casas' 'insulae' etc, to make sure a single house that keeps shifting states (especially for palaces), doesn't keep disappearing off the overlay.

rneeft commented:
Thank you for the review. Let me try making the overlay buttons dynamic, probably have time for that on Sunday.

I'll remove the new translations.

About the groups: I though about it however my problem was that I have all large insulae expect for one or two small and I wasn't able to find it.

rneeft committed:
Removed traslations

rneeft committed:
Removed translations
Removed TR as most as I could
Overhaul for overlay menu
- multiple sub menu support
- dynamic menu support

rneeft committed:
Removed changes in unaffected files.

rneeft committed:
Fixed issue with hovering overlay

rneeft committed:
Fixed minor issue - removing old code

rneeft commented:
@Sephirex95 I hope these changes are to your liking 🙂  

I initially experimented with a more complex, multi-layered menu, but it became too complicated. In the end, I settled on a structure that matches the existing building menus for consistency.  

I believe I’ve addressed all your requests regarding translations. For the few remaining cases, I wasn’t able to find a proper solution — suggestions are welcome! 

Here are some screenshots of the final result:
<img width="374" height="466" alt="start" src="https://github.com/user-attachments/assets/db83a724-8721-4fc6-a848-4e5e273dc402" />

<img width="374" height="466" alt="housing" src="https://github.com/user-attachments/assets/f2968bc9-7bd5-46f4-9b1f-84c7f1131237" />

<img width="374" height="466" alt="group" src="https://github.com/user-attachments/assets/502e6d4e-ae07-44b8-b5df-dca740a8711d" />

Sephirex95 commented:
> @Sephirex95 I hope these changes are to your liking 🙂
Looks great, thanks! I'll review the code tonight and hopefully we can merge soon 🤞

rneeft commented:
> Update: @rneeft There are some compiler errors for MSVC builds, could you please look into that?

Yes. I've seen the issues, I'm setting up a Windows machine as we speak to troubleshoot. 😅

rneeft committed:
Fixed compiler issue on Windows x64 systems.

Sephirex95 commented on src/widget/city_with_overlay.c:
```diff
@@ -35,6 +35,7 @@
 #include "widget/city_overlay_entertainment.h"
 #include "widget/city_overlay_health.h"
 #include "widget/city_overlay_other.h"
+#include "widget/city_overlay_housing.h"
```
I'll move this to be in the alphabetical order, just a note

Sephirex95 committed:
Stylistic changes and autoconfig

@rneeft implemented style changes I found on first glace

Sephirex95 commented:
Code's good, i just need to test it when I got a moment, @rneeft are you ready to merge? Are the style changes I added alright?

rneeft committed:
Merge branch 'master' into feature/housing_overlay

# Conflicts:
#	src/translation/english.c
#	src/translation/translation.h

rneeft commented:
What kind of overlay would you prefer @eugenikus8? a "dumb" overlay which just only shows the building of each god seperate, or a smart one in which it shows that god but also show a corresponding priest with a column on the house?  I'm not use the later is possible?

Sephirex95 commented:
nifty trick with translation types. 
I'll merge now, the religion overlay if you guys agree on that one, please make as separate PR, easier to review on it's own merit :)
